### PR TITLE
Linux Camera feature added

### PR DESCRIPTION
**Pull request by bhaveshAn**
This PR contains the take_picture and take_video methods to capture images and videos from command line and for different file formats as well.
However if no file format is specified then images will be saved in jpg format and video will be saved in avi format.

### CODE REVIEW

File: plyer/platforms/linux/camera.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    def _take_picture(self, on_complete, filename=None):
+        assert(on_complete is not None)
+        self.on_complete = on_complete
+        self.filename = filename
+        try:
+            name, extension = filename.split(".")
+        except:
+            extension = None
+        if extension = "jpg":
```
Comment (malverick) [inline plyer/platforms/linux/camera.py]: should be `==` instead `=`
causes syntax error at the time of installation

File: plyer/platforms/linux/camera.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            subprocess.call(["gst-launch", "v4l2src", "num-buffers=1", "!",
+                             "jpegenc", "!", "filesink",
+                             "location=" + self.filename])
+        elif extension = "png":
```
Comment (malverick) [inline plyer/platforms/linux/camera.py]: same here

File: plyer/platforms/linux/camera.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            subprocess.call(["gst-launch", "v4l2src", "num-buffers=1",
+                             "!", "pngenc", "!", "filesink",
+                             "location=" + self.filename])
+        elif extension = None:
```
Comment (malverick) [inline plyer/platforms/linux/camera.py]: same here

File: plyer/platforms/linux/camera.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    def _take_video(self, on_complete, filename=None):
+        assert(on_complete is not None)
+        self.on_complete = on_complete
+        self.filename = filename
+        try:
+            name, extension = filename.split(".")
+        except:
+            extension = None
+        if extension = None:
```
Comment (malverick) [inline plyer/platforms/linux/camera.py]: same here

**Comment by malverick** [comment]
@bhaveshAn after doing the above changes, logs generated after running main.py are:
```
[INFO   ] [Logger      ] Record log in /home/sumit/.kivy/logs/kivy_17-03-24_34.txt
[INFO   ] [Kivy        ] v1.9.2.dev0, git-ce9206c, 20170318
[INFO   ] [Python      ] v2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609]
[INFO   ] [Factory     ] 193 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_pygame, img_pil, img_gif (img_ffpyplayer ignored)
[INFO   ] [Text        ] Provider: pygame
[INFO   ] [Window      ] Provider: pygame(['window_egl_rpi'] ignored)
[INFO   ] [GL          ] Using the "OpenGL" graphics system
[INFO   ] [GL          ] Backend used <gl>
[INFO   ] [GL          ] OpenGL version <3.0 Mesa 12.0.6>
[INFO   ] [GL          ] OpenGL vendor <Intel Open Source Technology Center>
[INFO   ] [GL          ] OpenGL renderer <Mesa DRI Intel(R) HD Graphics 5500 (Broadwell GT2) >
[INFO   ] [GL          ] OpenGL parsed version: 3, 0
[INFO   ] [GL          ] Shading version <1.30>
[INFO   ] [GL          ] Texture max size <8192>
[INFO   ] [GL          ] Texture max units <32>
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [Clipboard   ] Provider: gtk3(['clipboard_dbusklipper'] ignored)
xclip version 0.12
Copyright (C) 2001-2008 Kim Saunders et al.
Distributed under the terms of the GNU GPL
[INFO   ] [Cutbuffer   ] Provider: xclip
[INFO   ] [CutBuffer   ] cut buffer support enabled
[INFO   ] [GL          ] NPOT texture support is available
[INFO   ] [OSC         ] using <multiprocessing> for socket
[INFO   ] [Base        ] Start application main loop
[INFO   ] [Base        ] Leaving application in progress...
 Traceback (most recent call last):
   File "main.py", line 77, in <module>
     CameraDemoApp().run()
   File "/usr/local/lib/python2.7/dist-packages/kivy/app.py", line 828, in run
     runTouchApp()
   File "/usr/local/lib/python2.7/dist-packages/kivy/base.py", line 504, in runTouchApp
     EventLoop.window.mainloop()
   File "/usr/local/lib/python2.7/dist-packages/kivy/core/window/window_pygame.py", line 403, in mainloop
     self._mainloop()
   File "/usr/local/lib/python2.7/dist-packages/kivy/core/window/window_pygame.py", line 289, in _mainloop
     EventLoop.idle()
   File "/usr/local/lib/python2.7/dist-packages/kivy/base.py", line 342, in idle
     self.dispatch_input()
   File "/usr/local/lib/python2.7/dist-packages/kivy/base.py", line 327, in dispatch_input
     post_dispatch_input(*pop(0))
   File "/usr/local/lib/python2.7/dist-packages/kivy/base.py", line 233, in post_dispatch_input
     listener.dispatch('on_motion', etype, me)
   File "kivy/_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy/_event.c:7726)
   File "/usr/local/lib/python2.7/dist-packages/kivy/core/window/__init__.py", line 1151, in on_motion
     self.dispatch('on_touch_down', me)
   File "kivy/_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy/_event.c:7726)
   File "/usr/local/lib/python2.7/dist-packages/kivy/core/window/__init__.py", line 1167, in on_touch_down
     if w.dispatch('on_touch_down', touch):
   File "kivy/_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy/_event.c:7726)
   File "/usr/local/lib/python2.7/dist-packages/kivy/uix/widget.py", line 457, in on_touch_down
     if child.dispatch('on_touch_down', touch):
   File "kivy/_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy/_event.c:7726)
   File "/usr/local/lib/python2.7/dist-packages/kivy/uix/widget.py", line 457, in on_touch_down
     if child.dispatch('on_touch_down', touch):
   File "kivy/_event.pyx", line 718, in kivy._event.EventDispatcher.dispatch (kivy/_event.c:7726)
   File "/usr/local/lib/python2.7/dist-packages/kivy/uix/behaviors/button.py", line 151, in on_touch_down
     self.dispatch('on_press')
   File "kivy/_event.pyx", line 714, in kivy._event.EventDispatcher.dispatch (kivy/_event.c:7681)
   File "kivy/_event.pyx", line 1225, in kivy._event.EventObservers.dispatch (kivy/_event.c:13524)
   File "kivy/_event.pyx", line 1109, in kivy._event.EventObservers._dispatch (kivy/_event.c:12356)
   File "/usr/local/lib/python2.7/dist-packages/kivy/lang/builder.py", line 64, in custom_callback
     exec(__kvlang__.co_value, idmap)
   File "/home/sumit/Downloads/plyer-linux_camera/examples/camera/basic/camerademo.kv", line 21, in <module>
     on_press: root.do_capture()
   File "main.py", line 40, in do_capture
     on_complete=self.camera_callback)
   File "/usr/local/lib/python2.7/dist-packages/plyer-1.2.5.dev0-py2.7.egg/plyer/facades/camera.py", line 60, in take_picture
     self._take_picture(filename=filename, on_complete=on_complete)
   File "/usr/local/lib/python2.7/dist-packages/plyer-1.2.5.dev0-py2.7.egg/plyer/platforms/linux/camera.py", line 18, in _take_picture
     "location=" + self.filename])
   File "/usr/lib/python2.7/subprocess.py", line 523, in call
     return Popen(*popenargs, **kwargs).wait()
   File "/usr/lib/python2.7/subprocess.py", line 711, in __init__
     errread, errwrite)
   File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child
     raise child_exception
 OSError: [Errno 2] No such file or directory
```
These logs are generated as soon as I press the `Take picture from camera!` button.

**Comment by bhaveshAn** [comment]
Tested and working fine on all versions of Linux Ubuntu.
However, this has not been merged so I am providing the link to [illustration of the feature](https://www.youtube.com/watch?v=jxTlNwbzJoY)
It does not having any dependency either. Hence in a merge-able state

File: plyer/facades/temperature.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 # coding=utf-8
 
+
```
Comment (KeyWeeUsr) [inline plyer/facades/temperature.py]: same

File: plyer/platforms/linux/camera.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import subprocess
+from plyer.facades import Camera
+
+
+class LinuxCamera(Camera):
```
Comment (KeyWeeUsr) [inline plyer/platforms/linux/camera.py]: Again, name it appropriately. It's GStreamer's functionality, which might be available even on Windows (not sure) :)

**Comment by bhaveshAn** [comment]
@phunsukwangdu If you have gone through the code of this PR. You may have come across the fact that `gst-launch` is not a pre-installed utility for linux. Infact `gst-launch-1.0` is the one such which does not need to be installed manually. And when to record the video gst-launch used `ffmpegcolorspace` which is not supported by gst-launch-1.0 instead using `videoconvert` which is best option.
Hence you submitted the PR which had dependencies on plyer #275 
And my PR worked on to find such utility that does not become a dependency for plyer.

**Comment by akshayaurora** [comment]
None of these pos address the issue that the the assumed binaries could not exist on a linux system.
We need something that is more generic and available on most linux systems.

**Comment by akshayaurora** [comment]
Assuming gstreamer  would be available on most linux systems would be a stretch, please look at using more generic methods like using v4l2 which has a larger penetration or something equivalent that would be guaranteed to have the binaries in the linux world.

**Comment by bhaveshAn** [comment]
@akshayaurora `gstreamer` is available in all linux distros. However the version can differ but i found `gst-launch-1.0` is being present in the Ubuntu 14.04 LTS and later versions.

*Realization behind using gstreamer*
The idea behind it is that when-ever a user wants to play media or record the media in a newly installed Ubuntu, the system installs the `gstreamer1.0-tools` package and its related plugins automatically.

Please look if something could happen, because we can use `whereis_exe` to give this feature on others linux distros as well, if it not run on other others.

**Pull request closed without merging**